lib/src/Status/Monitoring/Monitor.cpp

```diff
@@ -0,0 +1,60 @@
+#include "DOtherSide/Status/Monitoring/Monitor.h"
+
+#include <QCoreApplication>
+#include <QDebug>
+#include <QQmlApplicationEngine>
+#include <QQmlComponent>
+#include <QQuickWindow>
+
+void Monitor::initialize(QQmlApplicationEngine* engine) {
+    QObject::connect(engine, &QQmlApplicationEngine::objectCreated, this,
+                     [engine](QObject *obj, const QUrl &objUrl) {
+        if (!obj) {
+            qWarning() << "Error while loading QML:" << objUrl << "."
+                       << "Monitor initialization failed.";
+            return;
+        }
+
+        QQuickWindow* window = qobject_cast<QQuickWindow*>(obj);
+        QQmlComponent cmp(engine, QCoreApplication::applicationDirPath()
+                          + QStringLiteral("/../monitoring/Main.qml"), window);
```

Comment:
That looks like the wrong dependency direction. Would be great if DOtherSide doesn't make any assumptions about clients. In this case, it requires clients to have `"/../monitoring/Main.qml"`.

What would you say about passing the entry point through the definition provided by clients?
```suggestion
        QQmlComponent cmp(engine, QCoreApplication::applicationDirPath()
                          + MONITORING_ENTRY_POINT, window);
```

Reply:
Yeah, I know, dependency direction is wrong, I didn't feel comfortable about that but just didn't have any better idea. But yeah, I like your proposition, I will try to pass it from Makefile, through CMakeLists.txt to the source. Thanks!